minecraft/src/main/kotlin/org/polyfrost/oneconfig/internal/compat/MoulConfigCompat.kt

```diff
@@ -0,0 +1,202 @@
+
+            is Accessor_GuiOptionEditorDropdown -> {
+                fun getIndex(): Int {
+                    val selectedObject: Any? = children.get()
+                    if (selectedObject == null) return -1
+
+                    return if (editor.`oneconfig$useOrdinal`()) {
+                        selectedObject as Int
+                    } else if (editor.`oneconfig$constants`() != null) {
+                        (selectedObject as Enum<*>).ordinal
+                    } else {
+                        editor.`oneconfig$values`().indexOf(selectedObject)
+                    }
+                }
+
+                fun setIndex(index: Int) {
+                    if (editor.`oneconfig$constants`() != null) {
+                        children.set(editor.`oneconfig$constants`()[index])
+                    } else if (editor.`oneconfig$useOrdinal`()) {
+                        children.set(index)
+                    } else {
+                        children.set(editor.`oneconfig$values`()[index])
+                    }
+                }
+
+                property.getter = ::getIndex
+                property.setter = setter@{
+                    val index = it as? Int ?: return@setter
+                    setIndex(index)
+                }
+
+                property.metadata["options"] = editor.`oneconfig$values`()
+
+                DropdownVisualizer::class.java
+            }
+
+
+            is GuiOptionEditorSliderAccessor -> {
+                property.metadata["min"] = editor.`oneconfig$minValue`
+                property.metadata["max"] = editor.`oneconfig$maxValue`
+                property.getter = { (children.get() as? Number)?.toFloat() ?: editor.`oneconfig$maxValue` }
+                property.setter = setter@{ value ->
+                    val numberValue = value as? Number ?: return@setter
+                    fun isAny(type: Type, numberType: KClass<out Number>): Boolean {
+                        return numberType == type || numberType.java == type || numberType.javaObjectType == type || numberType.javaPrimitiveType == type
+                    }
+
+                    val type = children.type
+                    when {
+                        isAny(type, Int::class) -> children.set(numberValue.toInt())
+                        isAny(type, Float::class) -> children.set(numberValue.toFloat())
+                        isAny(type, Short::class) -> children.set(numberValue.toShort())
+                        isAny(type, Long::class) -> children.set(numberValue.toLong())
+                        isAny(type, Double::class) -> children.set(numberValue.toDouble())
+                        else -> null // do nothing/unknown number format?
+                    }
+                }
+
+                SliderVisualizer::class.java
+            }
+
+            is GuiOptionEditorInfoText -> return null
+            is GuiOptionEditorText -> TextVisualizer::class.java
+            is GuiOptionEditorDraggableList -> return null
```

Comment:
Should probably come back to these at some point and consider adding them to OneConfig(?), or close alternatives?